clients/client/src/main/java/org/projectnessie/client/http/impl/jdk11/JavaRequest.java

```diff
@@ -0,0 +1,431 @@
+
+      if (response.statusCode() >= 400) {
+        throw new HttpClientException(
+            String.format(
+                "%s request to %s failed with HTTP/%d", method, uri, response.statusCode()));
+      }
+
+      response = null;
+      return new org.projectnessie.client.http.HttpResponse(responseContext, config.getMapper());
+    } finally {
+      if (response != null) {
+        try {
+          LOGGER.debug(
+              "Closing unprocessed input stream for {} request to {} delegating to {} ...",
+              method,
+              uri,
+              response.body());
+          response.body().close();
+        } catch (IOException e) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  /**
+   * Executor used to serialize the response object to JSON.
+   *
+   * <p>Java's new {@link HttpClient} uses the {@link Flow.Publisher}/{@link Flow.Subscriber}/{@link
+   * Flow.Subscription} mechanism to write and read request and response data. We have to use that
+   * protocol. Since none of the implementations must not block, writes and reads run in a separate
+   * pool.
+   *
+   * <p>Jackson has no "reactive" serialization mechanism, which means that we have to provide a
+   * custom {@link OutputStream}, which delegates {@link Flow.Subscriber#onNext(Object) writes} to
+   * the subscribing code.
+   */
+  private static final Executor writerPool =
+      new ForkJoinPool(Math.max(8, ForkJoinPool.getCommonPoolParallelism()));
+
+  private void writeToOutputStream(RequestContext context, OutputStream outputStream)
+      throws Exception {
+    Object body = context.getBody().orElseThrow(NullPointerException::new);
+    try {
+      try (OutputStream out = wrapOutputStream(outputStream)) {
+        writeBody(config, out, body);
+      }
+    } catch (JsonGenerationException | JsonMappingException e) {
+      throw new HttpClientException(
+          String.format(
+              "Cannot serialize body of %s request against '%s'. Unable to serialize %s",
+              context.getMethod(), context.getUri(), body.getClass()),
+          e);
+    }
+  }
+
+  /**
+   * {@link Submitter} performs asynchronous writes using a plain, old {@link OutputStream}.
+   *
+   * <p>Sadly Jackson only provides functionality to serialize to an {@link OutputStream} or the
+   * like, for which neither Java's {@link Flow} API nor the new HTTP client API have support for.
+   */
+  private final class Submitter extends OutputStream
+      implements Flow.Publisher<ByteBuffer>, Runnable, Flow.Subscription {
+
+    // Okay - this is weird - but it is necessary when running tests with Quarkus via
+    // `./gradlew :nessie-quarkus:test`.
+    private final ClassLoader cl = Thread.currentThread().getContextClassLoader();
+    private final RequestContext context;
+    private final AtomicBoolean subscribed = new AtomicBoolean();
+    private final AtomicLong demand = new AtomicLong();
```

Comment:
Could we reuse `BodyPublishers.ofInputStream()` and set up a `java.nio.channels.Pipe` between Jackson output and HTTP input streams? I suppose it might be simpler in code and will probably provide similar runtime behaviour?

Reply:
Sorry, I'm confused... "unrelated issues" still prevented that approach? I'm just curious why it did not work :)

Reply:
Oh, I see you changed to `BodyPublishers.ofInputStream`... will re-review soon (hopefully)